src/component/legend/LegendView.js

```diff
@@ -361,9 +452,52 @@ export default echarts.extendComponentView({
         );
 
         var contentRect = contentGroup.getBoundingRect();
-        contentGroup.attr('position', [-contentRect.x, -contentRect.y]);
+        var contentPos = [-contentRect.x, -contentRect.y];
+
+        if (selectorGroup) {
+            // Place buttons in selectorGroup
+            selectorGroup && layoutUtil.box(
+                // Buttons in selectorGroup always layout horizontally
+                'horizontal',
+                selectorGroup,
+                legendModel.get('selectorItemGap', true)
+            );
+
+            var selectorRect = selectorGroup.getBoundingRect();
+            var selectorPos = [-selectorRect.x, -selectorRect.y];
+
+            var selectorLegendGap = retrieve2(
+                legendModel.get('selectorLegendGap', true), legendModel.get('itemGap', true)
+            );
+
+            var orientIdx = legendModel.getOrient().index;
+            var WH = orientIdx === 0 ? 'width' : 'height';
+            var HW = orientIdx === 0 ? 'height' : 'width';
+            var YX = orientIdx === 0 ? 'y' : 'x';
+            var selectorBoundaryGap = legendModel.get('selectorBoundaryGap', true);
+
+            if (selectorPosition === 'end') {
+                selectorPos[orientIdx] += contentRect[WH] + selectorLegendGap;
+            }
+            else {
+                contentPos[orientIdx] += selectorRect[WH] + selectorLegendGap;
```

Comment:
Hence "content" has been layout and positioned, 
I think we'd better do not modify contentPos (it makes complexity in complex content like scrollable legend view), but locate selector based on "content".

em... I will discuss with you about it later, after the final style confirmed.